lambdas/retrieve-cri-oauth-access-token/src/main/java/uk/gov/di/ipv/core/retrievecrioauthaccesstoken/RetrieveCriOauthAccessTokenHandler.java

```diff
@@ -151,7 +152,17 @@ public APIGatewayProxyResponseEvent handleRequest(
 
                 if (configurationService.isNotRunningInProd()) {
                     LOGGER.info("Submitting VC to CI storage system");
-                    submitVCAndSwallowErrors(vc, clientSessionDetailsDto.getGovukSigninJourneyId());
+                    try {
+                        ciStorageService.submitVC(
+                                vc, clientSessionDetailsDto.getGovukSigninJourneyId());
+                    } catch (Exception e) {
+                        LOGGER.error("Exception thrown when calling CI storage system", e);
+                        LOGGER.info(
+                                "Dropping VC due to exception thrown when calling CI storage system");
+                        throw new CredentialIssuerException(
+                                HTTPResponse.SC_SERVER_ERROR,
+                                ErrorResponse.FAILED_TO_SAVE_CREDENTIAL);
+                    }
```

Comment:
I think you're getting a codesmell cos this is nested try/catch. You might need to refactor it to a private method. This is why the `submitVCAndSwallowErrors` method existed.

Reply:
Should `ErrorResponse.FAILED_TO_SAVE_CREDENTIAL` be `ErrorResponse.ERROR_CALLING_CI_STORAGE`?